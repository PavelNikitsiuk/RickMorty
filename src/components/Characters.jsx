import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import style from './Characters.module.css';

class Characters extends PureComponent {
	static propTypes = {
		characters: PropTypes.arrayOf(
			PropTypes.shape({
				name: PropTypes.string,
				image: PropTypes.string,
			}),
		)
	};

	render() {
		const { characters } = this.props;

		return (
			<div className={style.container}>
				{characters.map((character, id) => (
					<Character key={id} character={character}/>
				))}
			</div>
		);
	}
}

export default Characters;
