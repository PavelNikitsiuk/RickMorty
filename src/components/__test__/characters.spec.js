import React from 'react'
import Characters from "../Characters";
import { shallow } from 'enzyme'

describe('characters', () => {
	const props = {
		characters: [
			{
				name: "string1",
				image: "src1"
			},
			{
				name: "string2",
				image: "src2"
			}
		]
	};

	const shallowCharacters = shallow(<Characters {...props} />);

	it('Characters renders properly', () => {
		expect(shallowCharacters).toMatchSnapshot();
	});
});