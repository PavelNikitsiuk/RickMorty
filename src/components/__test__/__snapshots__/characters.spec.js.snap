// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characters Character renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Characters
    characters={
      Array [
        Object {
          "image": "src1",
          "name": "string1",
        },
        Object {
          "image": "src2",
          "name": "string2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Character)
          character={
            Object {
              "image": "src1",
              "name": "string1",
            }
          }
        />,
        <withRouter(Character)
          character={
            Object {
              "image": "src2",
              "name": "string2",
            }
          }
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "character": Object {
            "image": "src1",
            "name": "string1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "character": Object {
            "image": "src2",
            "name": "string2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Character)
            character={
              Object {
                "image": "src1",
                "name": "string1",
              }
            }
          />,
          <withRouter(Character)
            character={
              Object {
                "image": "src2",
                "name": "string2",
              }
            }
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "character": Object {
              "image": "src1",
              "name": "string1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "character": Object {
              "image": "src2",
              "name": "string2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`characters Characters renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Characters
    characters={
      Array [
        Object {
          "image": "src1",
          "name": "string1",
        },
        Object {
          "image": "src2",
          "name": "string2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Character)
          character={
            Object {
              "image": "src1",
              "name": "string1",
            }
          }
        />,
        <withRouter(Character)
          character={
            Object {
              "image": "src2",
              "name": "string2",
            }
          }
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "character": Object {
            "image": "src1",
            "name": "string1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "character": Object {
            "image": "src2",
            "name": "string2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Character)
            character={
              Object {
                "image": "src1",
                "name": "string1",
              }
            }
          />,
          <withRouter(Character)
            character={
              Object {
                "image": "src2",
                "name": "string2",
              }
            }
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "character": Object {
              "image": "src1",
              "name": "string1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "character": Object {
              "image": "src2",
              "name": "string2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
